<a href="https://colab.research.google.com/github/cynthiiaa/MathStats/blob/main/ValidateWithInverseTransformSampling.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

I'll be validaing my answers to homework problems 4 and 6 using Inverse Transform Sampling.

<br>For problem 4, the pdf is given as $$f_Y\left(y\right)=2y$$
The cdf is $$F_Y\left(y\right)=y^2$$
The inverse of the cdf is $$\sqrt{x}$$
My theoretical result is $$ P(Y'_1 <0.6< Y'_5) ≈ 0.887 $$

In [ ]:
import math, random
import matplotlib.pyplot as plt
import numpy as np

def inverse_cdf1(x):
    return math.sqrt(x)

# generate a random number, u, from the standard uniform distribution in interval [0,1]
def sample_distribution(inverse_func):
    u = random.random()
    return inverse_func(u)

In [ ]:
def validate_probability(trials=1000):
  summation = 0
  for i in range(trials):
    y = [sample_distribution(inverse_cdf1) for i in range(5)]
    y.sort()
    if y[0] < 0.6 and y[4] > 0.6:
      summation += 1
  return summation/trials

In [ ]:
print(validate_probability())

0.884


**As you can see, 0.884 is very close to 0.887**

For problem 10, the pdf is given as $$f_Y\left(y\right)=e^{-y}$$
The cdf is $$F_Y\left(y\right)=1-e^{-y}$$
The inverse of the cdf is $$-\ln\left(1-x\right)$$
<br>My theoretical result is that **n** needs to be at least 12 in order for this condition to hold $$P\left(Y_{\min}<0.2\right)>0.9$$

In [ ]:
def inverse_cdf2(x):
    return -np.log(1-x)

def validate_n(trials=1000, n=12):
  summation = 0
  for i in range(trials):
    y = [sample_distribution(inverse_cdf2) for i in range(n)]
    y.sort()
    if y[0] < 0.2:
      summation += 1
  return summation/trials

In [ ]:
print(validate_n())

0.906


**As you can see, 0.906 is very close to 0.9**